projector-agent/build.gradle.kts

```
@@ -0,0 +1,142 @@
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact JetBrains, Na Hrebenech II 1718/10, Prague, 14000, Czech Republic
+ * if you need additional information or have any questions.
+ */
+
+plugins {
+  kotlin("jvm")
+  `maven-publish`
+}
+
+publishing {
+  publications {
+    create<MavenPublication>("maven") {
+      from(components["java"])
+    }
+  }
+}
+
+kotlin {
+  explicitApi()
+}
+
+val projectorClientGroup: String by project
+val projectorClientVersion: String by project
+val javassistVersion: String by project
+val agentClassName = "org.jetbrains.projector.agent.MainAgent"
+val launcherClassName = "org.jetbrains.projector.agent.AgentLauncher"
+
+
+dependencies {
+  implementation("$projectorClientGroup:projector-common:$projectorClientVersion")
+  implementation("$projectorClientGroup:projector-server-core:$projectorClientVersion")
+  implementation("$projectorClientGroup:projector-util-logging:$projectorClientVersion")
+  api(project(":projector-awt"))
+  api(project(":projector-server"))
+  implementation("org.javassist:javassist:$javassistVersion")
+}
+
+val serverTargetClasspath: Properties? = null
+val serverClassToLaunch: Properties? = null
+val ideaPath: Properties? = null
+
+if (project.rootProject.file("local.properties").canRead()) {
+  serverTargetClasspath?.loadPropertyFromResources("local.properties", "projectorLauncher.targetClassPath")
+  serverClassToLaunch?.loadPropertyFromResources("local.properties", "projectorLauncher.classToLaunch")
+  ideaPath?.loadPropertyFromResources("local.properties", "projectorLauncher.ideaPath")
+}
+
+tasks.jar {
+  manifest {
+    attributes(
+      "Can-Redefine-Classes" to true,
+      "Can-Retransform-Classes" to true,
+      "Agent-Class" to agentClassName,
+      "Main-Class" to launcherClassName,
+    )
+  }
+
+  exclude("META-INF/versions/9/module-info.class")
+
+  from({
+         configurations.runtimeClasspath.get().map { if (it.isDirectory) it else zipTree(it) }
+       })
+}
+
+println("----------- Agent launch config ---------------")
+println("Classpath: $serverTargetClasspath")
+println("ClassToLaunch: $serverClassToLaunch")
+println("------------------------------------------------")
+if (serverTargetClasspath != null && serverClassToLaunch != null) {
+  tasks.run {
+    group = "projector"
+    //classpath(sourceSets.main.runtimeClasspath, jar, )
+  }
+  (tasks["runWithAgent"] as JavaExec).apply {
```

This looks like using an already created task, but we need to create a new task here. It should look like this:
```suggestion
  tasks.create<JavaExec>("runWithAgent") {
```